static/app/components/forms/compactSelect.tsx

```diff
@@ -0,0 +1,316 @@
+import {Fragment, useEffect, useRef, useState} from 'react';
+import {components as selectComponents} from 'react-select';
+import styled from '@emotion/styled';
+import {useButton} from '@react-aria/button';
+import {FocusScope} from '@react-aria/focus';
+import {useMenuTrigger} from '@react-aria/menu';
+import {
+  AriaPositionProps,
+  OverlayProps,
+  useOverlay,
+  useOverlayPosition,
+} from '@react-aria/overlays';
+import {mergeProps} from '@react-aria/utils';
+import {useMenuTriggerState} from '@react-stately/menu';
+
+import Badge from 'sentry/components/badge';
+import Button from 'sentry/components/button';
+import DropdownButton, {DropdownButtonProps} from 'sentry/components/dropdownButtonV2';
+import SelectControl, {ControlProps} from 'sentry/components/forms/selectControl';
+import overflowEllipsis from 'sentry/styles/overflowEllipsis';
+import space from 'sentry/styles/space';
+
+type TriggerProps = {
+  props: Omit<React.HTMLAttributes<Element>, 'children'> & {
+    onClick?: (e: MouseEvent) => void;
+  };
+  ref: React.RefObject<HTMLButtonElement>;
+};
+
+type Props = ControlProps &
```

Comment:
Same here, I think you actually want to use interface and extend the different props

```ts
interface Props extends ControlProps, Partial<OverlayProps>...

```

The main reason for that is that if you have an overlapping type, the union will merge the type definition and narrow down your type and may end up in never type.

Consider this overlap
```ts
type PropsA = {
    value: number | null
}

type PropsB = {
    value: string | boolean
}

type FinalProps = PropsA & PropsB

const props: FinalProps = {
    value: "test"
}
```

^^ will error out with `Type 'string' is not assignable to type 'never'.` whereas using an interface like

```ts
interface PropsA {
    value: number | null
}

interface PropsB {
    value: string | boolean
}

interface FinalProps extends PropsA, PropsB {}
```

Will properly error out with 
```
Interface 'FinalProps' cannot simultaneously extend types 'PropsA' and 'PropsB'.
 Named property 'value' of types 'PropsA' and 'PropsB' are not identical.(2320)
 ```